lib/git-util.js

```diff
@@ -18,6 +18,67 @@ exports.latestTagOrFirstCommit = async () => {
 };
 
 exports.hasUpstream = async () => {
-	const {stdout} = await execa('git', ['status', '--short', '--branch', '--porcelain=2']);
+	const stdout = await execa.stdout('git', ['status', '--short', '--branch', '--porcelain=2']);
 	return /^# branch\.upstream [\w\-/]+$/m.test(stdout);
 };
+
+exports.currentBranch = () => execa.stdout('git', ['symbolic-ref', '--short', 'HEAD']);
+
+exports.isWorkingTreeClean = async () => {
+	try {
+		const status = await execa.stdout('git', ['status', '--porcelain']);
+		if (status !== '') {
+			return false;
+		}
+
+		return true;
+	} catch (_) {
+		return false;
+	}
+};
+
+exports.remoteHistoryClean = async () => {
+	let stdout;
+	try { // Gracefully handle no remote set up.
+		stdout = await execa.stdout('git', ['rev-list', '--count', '--left-only', '@{u}...HEAD']);
+	} catch (_) {}
+
+	if (stdout && stdout !== '0') {
+		return false;
+	}
+
+	return true;
+};
+
+exports.validRemote = async () => {
+	try {
+		await execa('git', ['ls-remote', 'origin', 'HEAD']);
+		return true;
+	} catch (error) {
+		return error.stderr.replace('fatal:', 'Git fatal error:');
+	}
+};
+
+exports.fetchRemote = () => execa('git', ['fetch']);
```

Comment:
The reason I added it is to avoid possible issues where `fetch` is already set to something like [the `fetch` function](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API). I guess that shouldn't be much of a problem here though, since this is a CLI for developers which won't get transpiled or anything.

Reply:
It's especially not a problem now that we import and use it as `git.fetch()`.